### PR TITLE
New gems & pickaxe

### PR DESCRIPTION
**Pull request by nekomaster1000**
Builds off of #40, so be sure to approve that before this one!

### CODE REVIEW

File: src/main/java/org/infernalstudios/miningmaster/config/MiningMasterConfig.java [hunk 1]
```diff
@@ -78,6 +81,21 @@ public MiningMasterConfig(ForgeConfigSpec.Builder builder) {
                 .translation(MiningMaster.MOD_ID + ".config.tooltip.diveAquamarineEnabled")
                 .define("diveAquamarineEnabled", true);
 
+        divineBerylEnabled = builder
+                .comment("Determines if Divine Beryl will generate.")
+                .translation(MiningMaster.MOD_ID + ".config.tooltip.divineBerylEnabled")
+                .define("diveAquamarineEnabled", true);
```
Comment (Hellion-Games) [inline src/main/java/org/infernalstudios/miningmaster/config/MiningMasterConfig.java]: Didn't change copy paste

File: src/main/java/org/infernalstudios/miningmaster/config/MiningMasterConfig.java [hunk 2]
```diff
@@ -78,6 +81,21 @@ public MiningMasterConfig(ForgeConfigSpec.Builder builder) {
...
+
+        spiderKunziteEnabled = builder
+                .comment("Determines if Spider Kunzite will generate.")
+                .translation(MiningMaster.MOD_ID + ".config.tooltip.spiderKunziteEnabled")
+                .define("diveAquamarineEnabled", true);
```
Comment (Hellion-Games) [inline src/main/java/org/infernalstudios/miningmaster/config/MiningMasterConfig.java]: Didn't change copy paste

File: src/main/java/org/infernalstudios/miningmaster/config/MiningMasterConfig.java [hunk 3]
```diff
@@ -78,6 +81,21 @@ public MiningMasterConfig(ForgeConfigSpec.Builder builder) {
...
+
+        unbreakingIoliteEnabled = builder
+                .comment("Determines if Unbreaking Iolite will generate.")
+                .translation(MiningMaster.MOD_ID + ".config.tooltip.unbreakingIoliteEnabled")
+                .define("diveAquamarineEnabled", true);
```
Comment (Hellion-Games) [inline src/main/java/org/infernalstudios/miningmaster/config/MiningMasterConfig.java]: Didn't change copy paste

File: src/main/java/org/infernalstudios/miningmaster/enchantments/FloatationEnchantment.java [hunk 4]
```diff
@@ -77,7 +77,7 @@ public void doPostAttack(LivingEntity user, Entity target, int level) {
         super.doPostAttack(user, target, level);
         if (target instanceof LivingEntity) {
             LivingEntity livingTarget = (LivingEntity) target;
-            livingTarget.addEffect(new MobEffectInstance(MobEffects.LEVITATION, 20 * level));
+            livingTarget.addEffect(new MobEffectInstance(MobEffects.LEVITATION, 4 * level));
```
Comment (Hellion-Games) [inline src/main/java/org/infernalstudios/miningmaster/enchantments/FloatationEnchantment.java]: Why is this being reduced down to 1/5 of a second? That seems extremely low

File: src/main/java/org/infernalstudios/miningmaster/init/MMBlocks.java [hunk 5]
```diff
@@ -62,6 +69,9 @@ public class MMBlocks {
     public static final RegistryObject<Block> HASTE_PERIDOT_BLOCK = registerBlockWithDefaultItem("haste_peridot_block", () -> new Block(getProperties(Blocks.DIAMOND_BLOCK)));
     public static final RegistryObject<Block> LUCKY_CITRINE_BLOCK = registerBlockWithDefaultItem("lucky_citrine_block", () -> new Block(getProperties(Blocks.DIAMOND_BLOCK)));
     public static final RegistryObject<Block> DIVE_AQUAMARINE_BLOCK = registerBlockWithDefaultItem("dive_aquamarine_block", () -> new Block(getProperties(Blocks.DIAMOND_BLOCK)));
+    public static final RegistryObject<Block> DIVINE_BERYL_BLOCK = registerBlockWithDefaultItem("divine_beryl_block", () -> new GemOreBlock(getProperties(Blocks.DIAMOND_BLOCK)));
+    public static final RegistryObject<Block> SPIDER_KUNZITE_BLOCK = registerBlockWithDefaultItem("spider_kunzite_block", () -> new GemOreBlock(getProperties(Blocks.DIAMOND_BLOCK)));
+    public static final RegistryObject<Block> UNBREAKING_IOLITE_BLOCK = registerBlockWithDefaultItem("unbreaking_iolite_block", () -> new GemOreBlock(getProperties(Blocks.DIAMOND_BLOCK)));
```
Comment (Hellion-Games) [inline src/main/java/org/infernalstudios/miningmaster/init/MMBlocks.java]: These should be Blocks, not GemOreBlocks

File: src/main/resources/data/miningmaster/advancements/miningmaster/geo_dude.json [hunk 6]
```diff
@@ -91,13 +91,37 @@
...
         "items": [
           {
             "items": [
-              "miningmaster:kinetic_opal_block"
+              "miningmaster:divine_beryl_block"
+            ]
+          }
+        ]
+      }
+    },
```
Comment (Hellion-Games) [inline src/main/resources/data/miningmaster/advancements/miningmaster/geo_dude.json:105]: If I'm reading this correctly, you removed the Kinetic Opal Block from this advancement, why? I don't see it added anywhere else in this file so it seems to be gone altogether.

File: src/main/resources/data/miningmaster/advancements/miningmaster/the_collector.json [hunk 7]
```diff
@@ -91,6 +91,42 @@
...
+      "conditions": {
+        "items": [
+          {
+            "items": [
+              "miningmaster:unbreaking_iolite"
+            ]
+          }
+        ]
+      }
+    },
```
Comment (Hellion-Games) [inline src/main/resources/data/miningmaster/advancements/miningmaster/the_collector.json:129]: You added these, but didn't actually add them to the requirements for the advancement.

File: src/main/resources/data/miningmaster/advancements/recipes/forging/ultima_axe_4.json [hunk 8]
```diff
@@ -2,29 +2,17 @@
   "parent": "minecraft:recipes/root",
   "rewards": {
     "recipes": [
-      "miningmaster:forging/ultima_axe_4"
+      "miningmaster:forging/ultima_axe_5"
```
Comment (Hellion-Games) [inline src/main/resources/data/miningmaster/advancements/recipes/forging/ultima_axe_4.json]: Why did you change this to 5 in this file, the 4 file, but delete the 5 file?

File: src/main/resources/data/miningmaster/advancements/recipes/forging/ultima_pickaxe_1.json [hunk 9]
```diff
@@ -30,25 +30,37 @@
...
         "items": [
           {
             "items": [
-              "miningmaster:kinetic_opal"
+              "miningmaster:haste_peridot_block"
+            ]
+          }
+        ]
+      }
+    },
```
Comment (Hellion-Games) [inline src/main/resources/data/miningmaster/advancements/recipes/forging/ultima_pickaxe_1.json]: Why is this now checking for a whole block of Haste Peridot? We may have talked about this, but I forget.

File: src/main/resources/data/miningmaster/recipes/forging/ultima_pickaxe_1.json [hunk 10]
```diff
@@ -11,16 +11,19 @@
       "item": "miningmaster:lucky_citrine"
     },
     {
-      "item": "miningmaster:haste_peridot"
+      "item": "miningmaster:haste_peridot_block"
```
Comment (Hellion-Games) [inline src/main/resources/data/miningmaster/recipes/forging/ultima_pickaxe_1.json]: Why does this one take a block, recipe 2 takes a single gem, and it looks like recipe 3 doesn't take any haste peridot?

**Review by Dariensg — CHANGES_REQUESTED** [review]
Few things:
1. You didn't change the catalyst tag at all for the changed tools.
2. I don't think you changed the ultima recipes to take the new tools instead of/in addition to the old ones.
3. The ultima recipes as a whole look a little wrong to me. I pointed out some of the issues, but I think they should be gone through carefully to make sure everything is right.

Everything else I noticed should be specified in comments on the relevant sections. Hopefully I didn't miss anything!

File: src/main/java/org/infernalstudios/miningmaster/enchantments/FloatationEnchantment.java (hunk 4 above)
Comment (nekomaster1000) [reply src/main/java/org/infernalstudios/miningmaster/enchantments/FloatationEnchantment.java]: Explained over Discord, but for prosperity: Levitation completely disables a ground-bound melee opponent's ability to attack. 1 second of Levitation max is more than enough time to get consecutive bow or arrow strikes in on an opponent.

File: src/main/resources/data/miningmaster/advancements/miningmaster/geo_dude.json (hunk 6 above)
Comment (nekomaster1000) [reply src/main/resources/data/miningmaster/advancements/miningmaster/geo_dude.json:105]: Oh whoops! It looks like in copy-pasting for the other gems, I accidentally glazed over and replaced the last remaining KO entry on this one. My bad!

File: src/main/resources/data/miningmaster/advancements/recipes/forging/ultima_axe_4.json (hunk 8 above)
Comment (nekomaster1000) [reply src/main/resources/data/miningmaster/advancements/recipes/forging/ultima_axe_4.json]: That's... A good question! I don't know! I deleted the 5 file due to the loss of the Garnet Axe, I think, but I don't know why I made this odd mistake here.

File: src/main/resources/data/miningmaster/recipes/forging/ultima_pickaxe_1.json (hunk 10 above)
Comment (nekomaster1000) [reply src/main/resources/data/miningmaster/recipes/forging/ultima_pickaxe_1.json]: Recipe 2 takes three Peridots (which is still a mistake) whilst recipe 3 doesn't take any Peridots because the catalyst is a Peridot Pickaxe. I realise the block solution is a little clunky, so I've decided to rejig them again without the block.

**Review by Dariensg — APPROVED** [review]
LGTM
